src/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useRef, useEffect } from "react";
+import { QueryResult, OperationVariables } from "@apollo/client";
+import { TextField, FormControl, FormLabel, Box, Button } from "@mui/material";
+import { useRouter } from "next/router";
+
+type Props = {
+  handleSubmitSearch: (
+    query: string
+  ) => Promise<QueryResult<any, OperationVariables>>;
+};
+
+const SearchInput = ({ handleSubmitSearch }: Props): JSX.Element => {
+  const queryInputRef = useRef<HTMLInputElement>(null);
```

Comment:
jsDoc - explain why uncontrolled input vs controlled - is selected here

Reply:
applied in commit: https://github.com/Sergei29/secret-escapes-tech-test/pull/3/commits/ba87d5d994e87cf0e145d855a11fdf857540ed59